digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/LinearAssetService.scala

```diff
@@ -511,6 +511,44 @@ trait LinearAssetOperations {
       if (sideCodeChanged(linearAsset, persistedLinearAsset)) dao.updateSideCode(linearAsset.id, SideCode(linearAsset.sideCode))
     }
   }
+
+  /**
+    * Mark VALID_TO field of old asset to sysdate and create a new asset.
+    * Copy all the data from old asset except the properties that changed, modifiedBy and modifiedAt.
+    */
+  private def updateValueByExpiration(assetId: Long, valueToUpdate: Value, valuePropertyId: String, username: String): Option[Long] = {
+    //Get Old Asset
+    val oldAsset =
+      valueToUpdate match {
+        case NumericValue(intValue) =>
+          dao.fetchLinearAssetsByIds(Set(assetId), valuePropertyId).head
+        case TextualValue(textValue) =>
+          dao.fetchAssetsWithTextualValuesByIds(Set(assetId), valuePropertyId).head
+        case maintenanceRoad: MaintenanceRoad =>
+          dao.fetchMaintenancesByIds(valuePropertyId.toInt, Set(assetId)).head
+        case _ => return None
+      }
+
+    //Expire the old asset
+    dao.updateExpiration(assetId, expired = true, username)
+
+    //Create New Asset
+    val newAssetIDcreate = createWithoutTransaction(oldAsset.typeId, oldAsset.linkId, valueToUpdate, oldAsset.sideCode, oldAsset.startMeasure, oldAsset.endMeasure, username, vvhClient.createVVHTimeStamp(5))
+
+    //Update modifiedBy, modifiedAt and update createdBy, createdDateTime to have the same createdBy/createdDate as old asset
+    val assetsUpdated = Queries.updateAssetModifiedByCopyOldOne(newAssetIDcreate, oldAsset.createdBy,
+      oldAsset.createdDateTime, username).first
+
+    //Update modifiedBy from LRM_POSITION table to have the same modifiedBy as old asset
+    val assetsLrmPosUpdated = Queries.updateLrmPositionModifiedByCopyOldOne(newAssetIDcreate, oldAsset.geomModifiedDate).first
+
```

Comment:
I belive we can change the createWithoutTransaction to get the modifiedby and modifiedDate. So like this we don't need to have 2 extra updates to the database. 
And i belive we don't need to change the modified date in the old lrm_position.